exo/main.py

```diff
@@ -208,6 +219,59 @@ def clean_path(path):
         path = path.strip('Optional("').rstrip('")')
     return os.path.expanduser(path)
 
+async def hold_outstanding(node: Node):
```

Comment:
If we're going to use the `Node` type here then it should be updated to include `outstanding_requests`.
Alternatively, the `Node` abstract class seems a literal unnecessary and we can probably just get rid of it and rename `StandardNode` to `Node` since that's the only implementation and it's not really a clean abstraction (how the heck would someone write another `Node` implementation)

Reply:
Yea, I think there's a little bit of "premature abstract base class" syndrome here. It's a common illness of OO code. Happy to refactor that